Add Calculate and Operate Module

smcommits opened the pull request:
Added Calculate Module to handle the logic for buttonName

The calculate module takes data object and button name. The data object contains the 'total', 'next', and 'operate' keys.
The calculate module checks the button name and then respectively modifies the data object.

Added Operate module to handle the logic for mathematical operation

The Calculate module calls the operate method.

smcommits committed:
Add Calculate and Operate Module

Added Calculate Module to handle the logic for buttonName

The calculate module takes data object and button name. The data object contains the 'total', 'next', and 'operate' keys.
The calculate module checks the button name and then respectively modifies the data object.

Added Operate module to handle the logic for mathematical operation

The Calculate module calls the operate method.

smcommits committed:
Add Logic folder and move calculate and operate modules

Forison commented:
##  Status::Changes required❌

# Hello @smcommits 

#### You have done a good job on this one so far, your mode of implementing the logic is quite unique, I looking forward to to how it will turn up, there are a couple of issues we have to address to get this project approved😇.

* [ ] The application throws into error after I try to launch
* [ ] Let us handle the division of a number by zero in the operate.js

#### That is it, kindly make the changes and submit a re-review

#### Happy coding

Addo Forison

smcommits committed:
Fix linter errors

smcommits committed:
Add logic for handeling zero in operate.js

Forison commented on src/logic/operate.js:
```diff
@@ -0,0 +1,20 @@
+  const operatorMap = {
+    '+': Big.prototype.plus,
+    '-': Big.prototype.minus,
+    x: Big.prototype.times,
```
why this 
x: Big.prototype.times,
and this
'*': Big.prototype.times,
These two seem to perform the same function. Let us be more consistent, since our hash table is one with string as it keys, I suggest with stick to that ie`'x': Big.prototype.times,`.

Forison reviewed (changes requested):
# Status::Change required❌

#### Good job one of the issues raised in the previous review, the application no longer throw an error when we run💪.

Big.js will throw an error if we try to divide a number by zero, this will throw an error if the user perfoms such an operation, to resolve this  we need to make sure we return `undefined` when the denominator is zero. 
Looking at your code, we can say, if `two ===0  && operator=== '/' return 'undefined'`

#### That is it, kindly make the change and submit a re-review.

#### Happy coding👨‍💻

Addo Forison

smcommits commented:
@Forison Hey! Thank you for the feedback. The double use .times is because of multiple hash mapping. As I am planning to implement keyboard bases input as well, the user can either input asterisks or 'x' as a symbol for multiplication.

smcommits committed:
Add logic operator

afanimartin reviewed (approved):
## Approved 👏🏿 👏🏿 👏🏿 
Nice work making the requested change. Merge and proceed to the next milestone.


### Martin Afani
#### Slack: whiz25

smcommits committed:
Add clickHandler in App

Added a clickHandler in App.js that passed down to the button component. The button component calls the click handler by providing
the button name and changing the state in app.js component.

smcommits committed:
Fix stylelint
